lib/linode.rb

```diff
@@ -83,14 +83,25 @@ def send_request(action, data)
     reformat_response(response)
   end
 
-  protected
+  def fetch_api_key(options = {})
+    request = { :api_action => 'user.getapikey', :api_responseFormat => 'json', :username => username, :password => password }
+
+    if options.key?(:label)
+      request[:label] = options[:label]
+    end
 
-  def fetch_api_key
-    response = post(:api_action => 'user.getapikey', :api_responseFormat => 'json', :username => username, :password => password)
+    if options.key?(:expires)
+      expires = options[:expires]
+      request[:expires] = expires.nil? ? 0 : expires
```

Comment:
This weird juggle is so `expires: 0` acts as the API documentation state, an explicit `expires: nil` also means "doesn't expire", but not passing `expires` at all gets the default behavior documented.